src/test/java/org/broadinstitute/hellbender/engine/spark/datasources/VariantsSparkSinkUnitTest.java

```diff
@@ -0,0 +1,126 @@
+        assertSingleShardedWritingWorks(vcf, outputFile.getAbsolutePath());
+    }
+
+    @Test(dataProvider = "loadVariants", groups = "spark")
+    public void variantsSinkHDFSTest(String vcf, String outputFileExtension) throws IOException {
+        final String outputHDFSPath = MiniClusterUtils.getTempPath(cluster, outputFileName, outputFileExtension).toString();
+        Assert.assertTrue(BucketUtils.isHadoopUrl(outputHDFSPath));
+        assertSingleShardedWritingWorks(vcf, outputHDFSPath);
+    }
+
+    @Test(dataProvider = "loadVariants", groups = "spark")
+    public void testWritingToAnExistingFileHDFS(String vcf, String outputFileExtension) throws IOException {
+        final Path outputPath = MiniClusterUtils.getTempPath(cluster, outputFileName, outputFileExtension);
+        final FileSystem fs = outputPath.getFileSystem(new Configuration());
+        Assert.assertTrue(fs.createNewFile(outputPath));
+        Assert.assertTrue(fs.exists(outputPath));
+        assertSingleShardedWritingWorks(vcf, outputPath.toString());
+    }
+
+    @Test(dataProvider = "loadVariants", groups = "spark")
+    public void testWritingToFileURL(String vcf, String outputFileExtension) throws IOException {
+        String outputUrl = "file://" + createTempFile(outputFileName, outputFileExtension).getAbsolutePath();
+        assertSingleShardedWritingWorks(vcf, outputUrl);
+    }
+
+    private void assertSingleShardedWritingWorks(String vcf, String outputPath) throws IOException {
+        JavaSparkContext ctx = SparkContextFactory.getTestSparkContext();
+
+        VariantsSparkSource variantsSparkSource = new VariantsSparkSource(ctx);
+        JavaRDD<VariantContext> variants = variantsSparkSource.getParallelVariantContexts(vcf, null);
+        VCFHeader header = getHeader(vcf);
+
+        VariantsSparkSink.writeVariants(ctx, outputPath, variants, header);
+
+        JavaRDD<VariantContext> variants2 = variantsSparkSource.getParallelVariantContexts(outputPath, null);
+        final List<VariantContext> writtenVariants = variants2.collect();
+
+        VariantContextTestUtils.assertEqualVariants(readVariants(vcf), writtenVariants);
+    }
+
+    private VCFHeader getHeader(String vcf) {
+        final java.nio.file.Path vcfPath = asPath(vcf);
+        try (SeekableStream stream = new SeekablePathStream(vcfPath)) {
+            return VCFHeaderReader.readHeaderFrom(stream);
+        } catch (IOException e) {
+            throw new UserException("Failed to read VCF header from " + vcf + "\n Caused by:" + e.getMessage(), e);
+        }
+    }
+
+    private static java.nio.file.Path asPath(String path) {
+        URI uri = URI.create(path);
+        return uri.getScheme() == null ? Paths.get(path) : Paths.get(uri);
+    }
+
+    private List<VariantContext> readVariants(String vcf) throws IOException {
+        File actualVcf;
+        // work around TribbleIndexedFeatureReader not reading header from .bgz files
+        if (vcf.endsWith(".bgz")) {
```

Comment:
`.vcf.gz` is the more common extension for compressed vcfs -- should handle that as well.